modules/caddyhttp/map/caddyfile.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2015 Matthew Holt and The Caddy Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package maphandler
+
+import (
+	"github.com/caddyserver/caddy/v2/caddyconfig/httpcaddyfile"
+	"github.com/caddyserver/caddy/v2/modules/caddyhttp"
+)
+
+func init() {
+	httpcaddyfile.RegisterHandlerDirective("map", parseCaddyfile)
+}
+
+// parseCaddyfile sets up the handler for a map from
+// Caddyfile tokens. Syntax:
+//     The map takes a <source> variable and maps it into the <dest> variable. The mapping process
+//     will check the <source> variable for the first successful match against a list of regular expressions.
+//     If a successful match is found the <dest> variable will contain the <replacement> value.
+//     If no successful match is found and the <default> is specified then the <dest> will contain the <default> value.
```

Comment:
It looks like the docs got garbled here. (This isn't syntax doc.)

Reply:
Ha yeah I dunno what the gdoc approach is but I have had a crack at fitting in with the others